app/tests/unit/lib/theme/widgets/DueDateWidget.test.tsx

```diff
@@ -0,0 +1,72 @@
+import DueDateWidget from "lib/theme/widgets/DueDateWidget";
+import { render, screen, fireEvent } from "@testing-library/react";
+import { DateTime } from "luxon";
+import getTimestamptzFromDate from "lib/helpers/getTimestamptzFromDate";
+import userEvent from "@testing-library/user-event";
+
+describe("The DueDateWidget", () => {
+  it("does not display a date when none is selected", () => {
+    const props: any = {
+      id: "test-id",
+      onChange: jest.fn(),
+      label: "Due date",
+      value: undefined,
+      required: true,
+    };
+    render(<DueDateWidget {...props} />);
+    expect(screen.getByText(/select a date/i)).toBeInTheDocument();
+  });
+
+  it("calls onChange with the correct date", () => {
+    const expectedDate = "2050-12-15T23:59:59.999-08:00";
+    const handleOnChange = jest.fn();
+    const props: any = {
+      id: "test-id",
+      onChange: handleOnChange,
+      label: "Due date",
+      value: "2020-01-01T23:59:59.999-07:00",
```

Comment:
The widget receives props as a string in this format, so I thought it might be better to use a string in the test than a calculation